worker/worker.go

```diff
@@ -164,7 +171,58 @@ func ParseWorkerConfiguration(reader io.Reader) Worker {
 	return configuration
 }
 
-func parseToken(tokenStr string) (map[string]interface{}, error){
+func (w *Worker) ExecTask(task *Task, serverEndPoint string) {
+	address := os.Getenv(WorkerNodeAddressKey)
+	client := utils.NewDockerClient(address)
+	taskExecutor := &TaskExecutor{Cli: *client}
+
+	stateChanges := make(chan TaskState)
+	go taskExecutor.Execute(task, stateChanges)
+
+	ticker := time.NewTicker(time.Duration(task.ReportInterval) * time.Second)
+
+	for {
+		select {
+		case <-ticker.C:
+			w.reportTask(task, taskExecutor, serverEndPoint)
+		case state := <-stateChanges:
+			task.State = state
+			ticker.Stop()
+			w.reportTask(task, taskExecutor, serverEndPoint)
+			return
+		}
+
+	}
+}
+
+func (w *Worker) reportTask(task *Task, executor *TaskExecutor, serverEndPoint string) {
+	executedCmdsLen, err := executor.Track()
+
+	if err != nil {
+		log.Println(err)
+	}
+
+	task.Progress = executedCmdsLen * 100 / len(task.Commands)
+
+	log.Println("progess: " + strconv.Itoa(task.Progress))
+
+	reportReq(w, task, serverEndPoint)
+}
+
+func reportReq(w *Worker, task *Task, serverEndPoint string) {
+	url := serverEndPoint + "/workers/" + w.Id + "/queues/" + w.QueueId + "/tasks"
+
+	header := http.Header{}
+	header.Set("arrebol-worker-token", w.Token)
+
+	resp, err := utils.Put(w.Id, task, header, url)
+
+	if err != nil || resp.StatusCode != 200 {
+		log.Println("Error on reporting task: " + err.Error())
+	}
+}
```

Comment:
The names of the current functions are not very clear. What do you think of refactoring theses lines with an `updateTask Progress` function to catch the progress of the task and `sendTaskReport` function to send the request.

Reply:
I've refactored a little taking your suggestion into account, could you check again please? I think it is better now, thank you!